src/useImageDimensions.ts

```diff
@@ -5,51 +5,55 @@ export interface URISource {
   uri: string
 }
 
-class Dimensions {
-  width: number
-  height: number
-  constructor(width: number, height: number) {
-    this.width = width
-    this.height = height
-  }
-  /**
-   * width to height ratio
-   */
-  get aspectRatio() {
-    return this.width / this.height
-  }
+export type Source = ImageRequireSource | URISource
+
+export interface ImageDimensions {
```

Comment:
Why do you also export this one?

Reply:
If someone wants to build a hook that extends on this hook, it's nice to be able to have one type that represents the return value. That way it can easily be used as the return value to a function and just passed on, e.g:

```typescript
function useLogoDimensions (): ImageDimensions {
  return useImageDimensions(globalSettings.logoImage)
}
```

Reply:
TypeScript calculates return types automatically, there is no need to mark it manually. 
Also, the 'ImageDimensions' name doesn't fully represent the content of this interface. There are 'error' and 'loading' fields. And the sizes themselves are even in a nested property 'dimensions', this field may be called 'ImageDimensions'. The whole object, which contains all of these - not really.

Reply:
I'd be open to renaming it `ImageDimensionsResult` or something similar...

> TypeScript calculates return types automatically, there is no need to mark it manually.

Many style guides mandate that the return type be specified to avoid accidentally not retuning what you intended by clearly stating the intent up front. I think that's a good practice 👍